src/wandbot/ingestion/preprocess_data.py

```diff
@@ -19,146 +19,210 @@
-        """Split text into chunks."""
-        text_splits = super().split_text(text)
-        text_splits = make_texts_tokenization_safe(text_splits)
-        return text_splits
-
-
-def convert_lc_to_llama(document: LcDocument) -> LlamaDocument:
-    """Converts a Langchain document to a Llama document.
-
-    Args:
-        document: A Langchain document.
-
-    Returns:
-        A Llama document.
-    """
-    llama_document = LlamaDocument.from_langchain_format(document)
-    excluded_embed_metadata_keys = [
-        "file_type",
-        "source",
-        "language",
-    ]
-    excluded_llm_metadata_keys = [
-        "file_type",
-    ]
-    llama_document.excluded_embed_metadata_keys = excluded_embed_metadata_keys
-    llama_document.excluded_llm_metadata_keys = excluded_llm_metadata_keys
-
-    return llama_document
-
-
-def load(documents: List[LcDocument], chunk_size: int = 512) -> List[Any]:
-    """Loads documents and returns a list of nodes.
-
-    Args:
-        documents: A list of documents.
-        chunk_size: The size of each chunk.
-
-    Returns:
-        A list of nodes.
-    """
-    md_parser = CustomMarkdownNodeParser(chunk_size=chunk_size)
-    code_parser = CustomCodeSplitter(
-        language="python", max_chars=chunk_size * 2
+    def transform_document(self, document: Document) -> Document:
+        return self.transform_documents([document])[0]
+
+
+def process_document_file(
+    documents: List[Document], transformer: DocumentTransformer
+) -> List[Document]:
+    transformed_documents = transformer.transform_documents(documents)
+
+    return list(transformed_documents)
+
+
+def load(
+    project: str,
+    entity: str,
+    source_artifact_path: str,
+    result_artifact_name: str = "transformed_data",
+) -> str:
+    run: wandb.wandb_sdk.wandb_run.Run = wandb.init(
+        project=project, entity=entity, job_type="preprocess_data"
+    )
+    artifact: wandb.Artifact = run.use_artifact(
+        source_artifact_path, type="dataset"
     )
-    # Define the node parser
-    node_parser = TokenTextSplitter.from_defaults(chunk_size=chunk_size)
+    artifact_dir: str = artifact.download()
 
-    llama_docs: List[LlamaDocument] = list(
-        map(lambda x: convert_lc_to_llama(x), documents)
+    document_files: List[pathlib.Path] = list(
+        pathlib.Path(artifact_dir).rglob("documents.jsonl")
     )
 
-    nodes: List[Any] = []
-    for doc in llama_docs:
-        try:
-            if doc.metadata["file_type"] == ".py":
-                parser = code_parser
-            else:
-                parser = md_parser
-            nodes.extend(parser.get_nodes_from_documents([doc]))
-        except Exception as e:
-            logger.error(f"Error parsing document: {e}")
-            logger.warning(
-                f"Unable to parse document: {doc.metadata['source']} with custom parser, using default "
-                f"parser instead."
+    lang_detect = FastTextLangDetect()
```

Comment:
i thought you mentioned we were removing langdetect? or is that the next update?